packages/azure-http-specs/specs/azure/resource-manager/resources/mockapi.ts

```diff
@@ -151,17 +163,121 @@ const validResourceExtensionsResource = {
   },
 };
 
+function getServiceKey(resourceUri: string) {
+  switch (resourceUri) {
+    case RESOURCE_GROUP_SCOPE_URI:
+      return "ResourceGroup" as const;
+    case SUBSCRIPTION_SCOPE_URI:
+      return "Subscription" as const;
+    case TENANT_SCOPE_URI:
+      return "Tenant" as const;
+    case RESOURCE_SCOPE_URI:
+      return "Resource" as const;
+    default:
+      throw new ValidationError(
+        "Invalid resource uri",
+        `${RESOURCE_GROUP_SCOPE_URI} | ${SUBSCRIPTION_SCOPE_URI} | ${TENANT_SCOPE_URI} | ${RESOURCE_SCOPE_URI}`,
+        resourceUri,
+      );
+  }
+}
+
+function requestHandler(
+  req: MockRequest,
+  requestMethod: string,
+  resourceUri: string,
+  validResource: any,
+) {
+  const serviceKey = getServiceKey(resourceUri);
+  switch (requestMethod) {
+    case "get":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json(validResource),
+      } as const;
+    case "put":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json(validResource),
+      } as const;
+    case "patch":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json({
+          ...validResource,
+          properties: {
+            provisioningState: "Succeeded",
+            description: "valid2",
+          },
+        }),
+      } as const;
+    case "delete":
+      return {
+        pass: serviceKey,
+        status: 204,
+      } as const;
+    case "list":
+      return {
+        pass: serviceKey,
+        status: 200,
+        body: json({
+          value: [validResource],
+        }),
+      } as const;
+    default:
+      throw new ValidationError(
+        "Invalid request method",
+        `"get" | "put" | "patch" | "delete" | "list"`,
+        requestMethod,
+      );
+  }
+}
+
+function getUri(resourceUri: string, requestMethod: string): string {
+  switch (requestMethod) {
+    case "list":
+      if (resourceUri === "/" || resourceUri === "") {
+        return `${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`;
+      }
+      return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`;
```

Comment:
Isn't this equivalent to just
```typescript
return `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`;
```
?

Reply:
This is not the case. When resourceUri is `/` and the URI format is `${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`, the returned URI is `//providers/Azure.ResourceManager.Resources/extensionsResources`, but if the URI format is `/${resourceUri}/providers/Azure.ResourceManager.Resources/extensionsResources`, the returned URI is `///providers/Azure.ResourceManager.Resources/extensionsResources`, which is not what the request requires.

Reply:
I see. Nit, better to use `"/" + TENANT_SCOPE_URI` and `TENANT_SCOPE_URI` here.